lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3232,6 +3233,74 @@ LogicalResult ConvertAtenOp<AtenGatherOp>::matchAndRewrite(
   return success();
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenIndexTensorOp>::matchAndRewrite(
+    AtenIndexTensorOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+  // Python example of this algorithm:
+  // https://gist.github.com/AmosLewis/c90c1148a96291db93408b3fa39f9ae2
+  auto input = adaptor.getSelf();
+  auto inputTensorType =
+      adaptor.getSelf().getType().dyn_cast<RankedTensorType>();
+  // Check input is a tensor type.
+  if (!inputTensorType)
+    return rewriter.notifyMatchFailure(
+        op, "Only tensor types input are currently supported");
+
+  // Deal with torch.prim.ListConstruct of non const value to get the index
+  auto tensorList = op.getIndices();
+  SmallVector<Value> tensorsTorchType;
+  if (!getListConstructElements(tensorList, tensorsTorchType))
+    return op.emitError(
+        "unimplemented: the tensor list is not from list construct");
+  auto tensors = getTypeConvertedValues(rewriter, op->getLoc(),
+                                        getTypeConverter(), tensorsTorchType);
+
+  auto index = tensors[0];
+  // TODO figure out why the index is empty for
+  // IndexTensorMultiInputContiguousCenter e2e test
+  if (!index.getImpl())
+    return rewriter.notifyMatchFailure(
+        op, "Only list ranked tensor types index are supported");
+  auto indexType = index.getType().dyn_cast<RankedTensorType>();
```

Comment:
I am pretty sure this will success. I have tried the dyn_cast<>, but it will segfault for [IndexTensorMultiInputContiguousCenter](https://gist.github.com/AmosLewis/144ff9fae33e1bf99cd3986326914cd4). Because to use dyn_cast, you will have to call index.getType(), which will get the *index.getImpl(). But sometimes it is void which will lead to segfault.

Reply:
example: torch.ops.aten.index(x, (None, index1, index2, None))